lib/src/editor.dart

```diff
@@ -0,0 +1,790 @@
+import 'package:yaml/src/equality.dart';
+import 'package:yaml/yaml.dart';
+
+import './source_edit.dart';
+
+/// An interface for modififying [YAML][1] documents while preserving comments and
+/// whitespaces.
+///
+/// YAML parsing is supported by `package:yaml`, and modifications are performed as
+/// string operations. Each time a modification takes place via one of the public
+/// methods, we calculate the expected final result, and parse the result YAML string,
+/// and ensure the two YAML trees match. Users may define the default settings to be
+/// applied to these string modifications. Note however that these settings only apply
+/// to portions of the YAML that are modified by this class.
+///
+/// Most modification methods require the user to pass in an [Iterable<Object>] path that
+/// holds the keys/indices to navigate to the element. Key equality is performed via
+/// `package:yaml`'s [YamlMap]'s key equality.
+///
+/// [1]: https://yaml.org/
+class YamlEditor {
```

Comment:
It would be cool to have functionality to `insertCommentBeforePath` or something in that style.
Not sure how hard that is 😊

Reply:
It would be cool! Added it as an issue #3, and tagged it to the [project board](https://github.com/walnutdust/yaml_edit/projects/1)